components/omega/src/ocn/TendencyTerms.cpp

```diff
@@ -83,6 +93,68 @@ Tendencies *Tendencies::get(const std::string &Name ///< [in] Name of tendencies
 
 } // end get tendencies
 
+//------------------------------------------------------------------------------
+// read and set config options
+int Tendencies::readTendConfig(Config *TendConfig ///< [in] Tendencies subconfig
+) {
+   int Err = 0;
+
+   Err = TendConfig->get("ThicknessFluxTendencyEnable",
+                         this->ThicknessFluxDiv.Enabled);
+   if (Err != 0) {
+      LOG_CRITICAL("Tendencies: ThicknessFluxTendencyEnable not found in "
+                   "TendConfig");
+      return Err;
+   }
+
+   Err = TendConfig->get("PVTendencyEnable", this->PotientialVortHAdv.Enabled);
+   if (Err != 0) {
+      LOG_CRITICAL("Tendencies: PVTendencyEnable not found in TendConfig");
+      return Err;
+   }
+
+   Err = TendConfig->get("KETendencyEnable", this->KEGrad.Enabled);
+   if (Err != 0) {
+      LOG_CRITICAL("Tendencies: KETendencyEnable not found in TendConfig");
+      return Err;
+   }
+
+   Err = TendConfig->get("SSHTendencyEnable", this->SSHGrad.Enabled);
+   if (Err != 0) {
+      LOG_CRITICAL("Tendencies: SSHTendencyEnable not found in TendConfig");
+      return Err;
+   }
+
+   Err = TendConfig->get("VelDiffTendencyEnable",
+                         this->VelocityDiffusion.Enabled);
+   if (Err != 0) {
+      LOG_CRITICAL("Tendencies: VelDiffTendencyEnable not found in TendConfig");
+      return Err;
+   }
+
+   I4 Err1 = TendConfig->get("ViscDel2", this->VelocityDiffusion.ViscDel2);
+   if (Err1 != 0 && this->VelocityDiffusion.Enabled) {
+      LOG_CRITICAL("Tendencies: ViscDel2 not found in TendConfig");
+      return Err1;
+   }
```

Comment:
Why are `ViscDel2` and `ViscDel4` using separate error code variables, but everything else uses `Err` ?

Reply:
These `get` calls should only generate an error that is returned if the relevant tendency term is enabled, but the viscosity coefficient isn't found. This really only matters with the last get for `ViscDel4`, since it's the final operation in the function; if `get(ViscDel4)` returned a non-zero error code and stored it in `Err`, this would be returned even if VelocityHyperDiff wasn't enabled. Looking at it again, I might just opt for separate error code variables for each `get` in the `readTendConfig` function, since it just looks weird for everything else to use the same `Err` and these two to be special.